### PR TITLE
feat: support for uploading folders in this current example

### LINKED ISSUE #376: :octocat:  Support for uploading and downloading folders
This current example at `graphql-server/go-server/examples` only supports the uploading and downloading of a single file, consider accomplishing the following two requirements:
- Support for uploading folders, keep the same organization in minio
- File upload problem, at present, if the file write absolute path, because there is no handling of the prefix `/`, resulting in storage failure in minio, that is to say, if the user uses the parameter `--file=/local/a.tx` , after we read the file, this path to be modified to `local/a.txt`

### PR DESCRIPTION
**Pull request by dayuy**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it
support for uploading folders in this current example

#### Which issue(s) this PR fixes

Fixes #376

### CODE REVIEW

**Comment by 0xff-dev** [comment]
https://github.com/kubeagi/arcadia/pull/410/files#diff-584560258ba40e19973b4a2ce0006d0be82697674e47353450eaaaa5ce4b9868L327

```go
if resp.StatusCode/100 != 2 {
	klog.Errorf("[do Error] expect 200 get %d", resp.StatusCode)
	return fmt.Errorf("not 200")
}
```

**Comment by dayuy** [comment]
> https://github.com/kubeagi/arcadia/pull/410/files#diff-584560258ba40e19973b4a2ce0006d0be82697674e47353450eaaaa5ce4b9868L327
> 
> ```go
> if resp.StatusCode/100 != 2 {
> 	klog.Errorf("[do Error] expect 200 get %d", resp.StatusCode)
> 	return fmt.Errorf("not 200")
> }
> ```

done

File: apiserver/examples/upload-download-file/main.go [hunk 1]
```diff
@@ -537,7 +545,25 @@ func main() {
 	tp := &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}}
 
 	if *action == upload {
-		uploadFile(*fileName, *bucket, *bucketPath, tp)
+		info, err := os.Stat(*fileName)
```
Comment (0xff-dev) [inline apiserver/examples/upload-download-file/main.go:548]:
```go
filepath.WalkDir(*fileName, func(path string, d fs.DirEntry, err error) error {
    if err != nil {
        klog.Errorf("[Error] failed access a path %s: %s", path, err)
        return err
    }
    if !d.IsDir() {
        uploadFile(path, *bucket, *bucketPath, tp)
    }
    return nil
})
```

Don't need the previous stat to check it's a file, just call `WalkDir`.

**Review by 0xff-dev — APPROVED** [review]

File: apiserver/examples/upload-download-file/main.go [hunk 2]
```diff
@@ -228,11 +231,11 @@ func genURL(
 		klog.Errorf("[Error] send genMultipartURL request error %s", err)
 		return GenChunkURLResult{}, err
 	}
+	defer resp.Body.Close()
```
Comment (bjwswang) [inline apiserver/examples/upload-download-file/main.go:234]: @dayuy @0xff-dev So this is only a client side example to upload folders ,right?

Comment (0xff-dev) [reply apiserver/examples/upload-download-file/main.go:234]: Yes. Our API can only do one file upload, to upload a directory you need the client to upload each file according to the directory structure.

**Pull request merged**